lib/gateway.js

```diff
@@ -281,6 +289,47 @@ async function buildGateway (gatewayOpts, app) {
     schema,
     serviceMap,
     entityResolversFactory: factory,
+    pollingInterval: gatewayOpts.pollingInterval,
+    async refresh () {
+      if (this._serviceSDLs === undefined) {
+        this._serviceSDLs = serviceSDLs.join('')
+      }
+
+      await allSettled(
+        Object.values(serviceMap).map((service) =>
+          service.refresh().catch((err) => {
+            errorHandler(err, service)
+          })
+        )
+      ).then((result) => {
+        result
+          .filter(({ status }) => status === 'rejected')
+          .forEach(({ reason }) => {
+            throw reason
+          })
```

Comment:
nit: we might be losing some errors here. Should we report all of them?

Reply:
Yea, I agree
I will throw an error with information about all errors